app/policies/case_assignment_policy.rb

```diff
@@ -29,11 +29,11 @@ def reimbursement?
   end
 
   def hide_contacts?
-    !record.active? && !record.hide_old_contacts? && admin_or_supervisor?
+    show_or_hide_contacts? && !record.hide_old_contacts?
   end
```

Comment:
Something about this really bothers me. This is used in one place. https://github.com/rubyforgood/casa/blob/d7061ea1c9540735038a594422643328e29fb78b/app/views/casa_cases/_volunteer_assignment.html.erb#L59

```rb
<% if (policy(assignment).hide_contacts?) %>
  <%= button_to("Hide Old Contacts",
            show_hide_contacts_case_assignment_path(assignment),
            method: :patch,
            class: "btn btn-outline-secondary text-wrap") %>
<% else %>
  <%= button_to("Show Old Contacts",
            show_hide_contacts_case_assignment_path(assignment),
            method: :patch,
            class: "btn btn-outline-warning text-wrap") %>
<% end %>

```
All it does is dictate whether the show or hide button appears.

But the current logic is incorrect it will show a hide button if the user doesn't have show_hide permissions, which doesn't make sense, instead it should not show a button. We don't actually see this play out because you can't reach that page without the needed permissions.

I feel like this method should not exist and instead the above code should just check `assignement.hide_old_contacts?`

Reply:
Fully agree with this. Made it so in 475615fda687a798b6859a48be29d396240e3a2c.

But that got me thinking about other places `hide_old_contacts?` is used...
As you noted, that button toggle was the only place the policy method was used.

But i'm also not seeing any other place that `CaseAssignment#hide_old_contacts?` is used. Does that mean that clicking show/hide old contacts has no actual impact? Where do we actually _hide_ old contacts?

Reply:
It looks like the **method** is not used outside of that but the value is used as part of a query.


The show view calls a method off a decorator

https://github.com/rubyforgood/casa/blob/e596c58eb56f8c9b20840b3f5c6ab9aec3ee0daa/app/views/casa_cases/show.html.erb#L135

```rb
<%= render(partial: "case_contacts/case_contact",
          collection: @casa_case.decorate
            .case_contacts_filtered_by_active_assignment_ordered_by_occurred_at
            .includes(contact_types: [:contact_type_group])
            .grab_all(current_user),
          as: :contact) %>
```

That method contains a query that scopes down based on `hide_old_contacts`

https://github.com/rubyforgood/casa/blob/e596c58eb56f8c9b20840b3f5c6ab9aec3ee0daa/app/decorators/casa_case_decorator.rb#L18

```rb
  def case_contacts_filtered_by_active_assignment_ordered_by_occurred_at
    object.case_contacts
      .joins("INNER JOIN case_assignments on case_assignments.casa_case_id = case_contacts.casa_case_id and case_assignments.volunteer_id = case_contacts.creator_id")
      .where("case_assignments.hide_old_contacts = false")
      .order(occurred_at: :desc)
  end
```